haskell/private/cc_libraries.bzl

```diff
@@ -95,7 +95,7 @@ def get_library_files(hs, cc_libraries_info, libraries_to_link, dynamic = False,
         pic = dynamic
 
     # PIC is irrelevant on static GHC.
```

Comment:
Not sure I understand this comment on reflection, given the way we use `-fPIC` and `-fexternal-dynamic-refs` in this PR -- @aherrmann thoughts and also whether we should elaborate on it?

Reply:
Good question. I looked into the history, I introduced that comment [for Windows](https://github.com/tweag/rules_haskell/commit/16ed109d4659dec9f403d24a0080cd262f12bf79#diff-5fd52f58c9c22b2fe4b00f1b9d279ad0R269-R270) and later [generalized it to static RTS](https://github.com/tweag/rules_haskell/commit/fb37a1e4377621d96213746a596a99fb7bef8017#diff-5fd52f58c9c22b2fe4b00f1b9d279ad0L255-R256). But, it doesn't look quite right, indeed. `pic_required` controls whether a `LibraryToLink.static_library` can be preferred over a `.dynamic_library` or if it has to be a `.pic_static_library`. Even with a static RTS pic should be required for GHCi or Template Haskell. I think the trouble here is that too often a static archive is only assigned to `.static_library` even if `.pic_static_library` would be appropriate ([e.g. here](https://github.com/tweag/rules_haskell/blob/eb16a5401770098a801775ea3a893b09cafe054c/haskell/private/haskell_impl.bzl#L553)). It seems a better approach would be to assign generated static libraries to `.pic_static_library` if `-fPIC` is set and remove `pic_required` here. But, that seems outside the scope of this PR.